core/iwasm/fast-jit/jit_frontend.c

```diff
@@ -487,13 +488,50 @@ get_ent_and_bw_cnts_reg(JitFrame *frame)
              * ent_and_br_cnts = (uint32*)mod_ent_and_br_cnts +
              *     cur_wasm_func_idx
              */
-            GEN_INSN(LDPTR, frame->ent_and_bw_cnts_reg, mod_ent_and_bw_cnts_reg,
+            GEN_INSN(LDPTR, frame->ent_and_br_cnts_reg, mod_ent_and_bw_cnts_reg,
                      offset_reg);
         }
     }
 
-    return frame->ent_and_bw_cnts_reg;
+    return frame->ent_and_br_cnts_reg;
+}
+
+void
+gen_increase_cnt_insn(JitFrame *frame, uint32 cnt_idx)
+{
+    JitCompContext *cc = frame->cc;
+    JitReg ent_and_br_cnts_reg = get_ent_and_br_cnts_reg(frame);
+    JitReg cnt_reg = jit_cc_new_reg_I32(cc);
+    JitReg cnt_inc_reg = jit_cc_new_reg_I32(cc);
+    JitReg offset_reg;
+
+    bh_assert(cnt_idx > 0);
+
+    if (UINTPTR_MAX == UINT64_MAX) {
+        offset_reg = jit_cc_new_reg_I64(cc);
+        /* uint32[cnt_idx] -> x bytes*/
+        GEN_INSN(SHL, offset_reg, NEW_CONST(I64, cnt_idx), NEW_CONST(I64, 2));
+    }
+    else {
+        offset_reg = jit_cc_new_reg_I32(cc);
+        /* uint32[cnt_idx] -> x bytes*/
+        GEN_INSN(SHL, offset_reg, NEW_CONST(I32, cnt_idx), NEW_CONST(I32, 2));
+    }
+
+    /* load -> increase -> store */
+    JitInsn *insn = GEN_INSN(LDI32, cnt_reg, ent_and_br_cnts_reg, offset_reg);
+    GEN_INSN(ADD, cnt_inc_reg, cnt_reg, NEW_CONST(I32, 1));
+    GEN_INSN(STI32, cnt_inc_reg, ent_and_br_cnts_reg, offset_reg);
+}
+
+void
+gen_increase_cnt_insn_here(JitFrame *frame)
+{
+    bh_assert(frame->cc->ent_and_br_cnts_idx > 0
+              && "ent_and_br_cnts_idx in JitCompContext should be > 0");
+    gen_increase_cnt_insn(frame, frame->cc->ent_and_br_cnts_idx++);
```

Comment:
The function name `gen_increase_cnt_insn_here` looks a little strange, `gen_increase_cnt_insn` isn't called by other places, how about merge them as `gen_increase_cnt_insn`?

Reply:
I am thinking about blocking `ent_and_br_cnts_idx++` from other components. but I can remove it f it does confuse